cmd/syncer/cmd/syncer.go

```diff
@@ -0,0 +1,164 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+
+	nscontroller "github.com/kcp-dev/kcp/pkg/reconciler/namespace"
+	"github.com/kcp-dev/kcp/pkg/syncer"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/util/sets"
+	genericapiserver "k8s.io/apiserver/pkg/server"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/component-base/logs"
+	"k8s.io/klog/v2"
+)
+
+const numThreads = 2
+
+func NewSyncerCommand() *cobra.Command {
+	options := NewDefaultOptions()
+
+	syncerCommand := &cobra.Command{
+		Use:   "syncer",
+		Short: "Synchronizes resources in `kcp` assigned to the clusters",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			options.SyncedResourceTypes = args
+
+			//validation
+			err := options.Validate()
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	//bind flags
+	options.BindFlags(syncerCommand.Flags())
+
+	return syncerCommand
+}
+
+func (options *Options) Validate() error {
+	err := options.Complete()
+	if err != nil {
+		return err
+	}
+
+	err = options.Run()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (options *Options) Complete() error {
+	if len(options.SyncedResourceTypes) == 0 {
+		options.SyncedResourceTypes = []string{"deployments.apps"}
+		klog.Infoln("No resource types provided; using defaults.")
+	}
+	klog.Infof("Syncing the following resource types: %s", options.SyncedResourceTypes)
+
+	// Create a client to dynamically watch "from".
+	if options.FromClusterName == "" {
+		return errors.New("--from_cluster is required")
+	}
+
+	var err error
+	options.fromConfig, err = clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		&clientcmd.ClientConfigLoadingRules{ExplicitPath: options.FromKubeconfig}, nil).ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	if options.ToKubeconfig != "" {
+		var toOverrides clientcmd.ConfigOverrides
+		if options.ToContext != "" {
+			toOverrides.CurrentContext = options.ToContext
+		}
+
+		options.toConfig, err = clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+			&clientcmd.ClientConfigLoadingRules{ExplicitPath: options.ToKubeconfig},
+			&toOverrides).ClientConfig()
+		if err != nil {
+			return err
+		}
+	} else {
+		options.toConfig, err = rest.InClusterConfig()
+	}
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (options *Options) Run() error {
+	ctx := genericapiserver.SetupSignalContext()
+
+	if err := syncer.StartSyncer(ctx, options.fromConfig, options.toConfig, sets.NewString(options.SyncedResourceTypes...), options.FromClusterName, options.PclusterID, numThreads); err != nil {
+		return err
+	}
+
+	<-ctx.Done()
+
+	return nil
+}
+
+type Options struct {
+	FromKubeconfig  string
+	FromClusterName string
+	ToKubeconfig    string
+	ToContext       string
+	PclusterID      string
```

Comment:
WorkloadClusterName

Reply:
Hey @sttts, I'm sorry, I'm new to KCP but if I'm correct do you mean ClusterName is the same as WorkloadClusterName? And should I change ```FromClusterName``` to ```FromWorkloadClusterName```?